neqo-http3/tests/non_ascii_headers.rs

```diff
@@ -84,104 +81,23 @@ fn non_ascii_utf8_header_echo() {
 
     let headers = response_headers.expect("No response headers received");
 
-    // Verify the echoed header contains the original non-UTF-8 byte
+    // Verify the echoed header contains the original data
     let echoed_header = headers
-        .find_header("x-echoed-data")
+        .find_header(response_header_name)
         .expect("Echoed header not found");
-    assert_eq!(echoed_header.value(), test_bytes.as_slice());
-    assert_eq!(echoed_header.value(), &[0xE4]);
+    assert_eq!(echoed_header.value(), test_data);
 }
 
 #[test]
-fn non_ascii_emoji_header_echo() {
-    // Connect a client and a server.
-    let mut client = default_http3_client();
-    let mut server = default_http3_server();
-    let out = test_fixture::connect_peers(&mut client, &mut server);
-    assert_eq!(server.process(out, now()).dgram(), None);
-
-    // Ignore all events so far.
-    drop(server.events());
-    drop(client.events());
+fn non_ascii_utf8_header_echo() {
```

Comment:
```suggestion
fn extended_ascii_non_utf8_header_echo() {
```

Wouldn't this be correct?